python/activator/middleware_interface.py

```diff
@@ -175,12 +166,17 @@ def prep_butler(self, visit: Visit) -> None:
 
                 # CHAINED collections
                 export.saveCollection("refcats")
+                export.saveCollection("templates")
+                export.saveCollection(self.instrument.makeCollectionName("defaults"))
 
             self.butler.import_(filename=export_file.name,
                                 directory=self.central_butler.datastore.root,
                                 transfer="copy")
 
+        # Ensure all components of the instrument are registered.
+        self.instrument.register(self.butler.registry)
```

Comment:
I need this here for ingest to work. Without it I get:

```
lsst.daf.butler.registry._exceptions.DataIdValueError: Could not fetch record for required dimension detector via keys {'instrument': 'DECam', 'detector': 1, 'exposure': 1}.
```

at `self.rawIngestTask.run()`.

Reply:
Ah, because of [this issue](https://github.com/lsst/ap_verify_ci_hits2015/blob/main/scripts/make_preloaded_export.py#L46-L47). Can you clarify the comment to say that, depending on the dataset, the exported data doesn't necessarily include all detectors?